checker/jdk/nullness/src/javax/xml/parsers/FactoryFinder.java

```diff
@@ -0,0 +1,319 @@
+     *
+     * @param type                  Base class / Service interface  of the
+     *                              factory to find.
+     * @param fallbackClassName     Implementation class name, if nothing else
+     *                              is found.  Use null to mean no fallback.
+     *
+     * Package private so this code can be shared.
+     */
+    static <T> T find(Class<T> type, @Nullable String fallbackClassName)
+        throws FactoryConfigurationError
+    {
+        final String factoryId = type.getName();
+        dPrint("find factoryId =" + factoryId);
+
+        // Use the system property first
+        try {
+            String systemProp = ss.getSystemProperty(factoryId);
+            if (systemProp != null) {
+                dPrint("found system property, value=" + systemProp);
+                return newInstance(type, systemProp, null, true);
+            }
+        }
+        catch (SecurityException se) {
+            if (debug) se.printStackTrace();
+        }
+
+        // try to read from $java.home/lib/jaxp.properties
+        try {
+            if (firstTime) {
+                synchronized (cacheProps) {
+                    if (firstTime) {
+                        String configFile = ss.getSystemProperty("java.home") + File.separator +
+                            "lib" + File.separator + "jaxp.properties";
+                        File f = new File(configFile);
+                        firstTime = false;
+                        if (ss.doesFileExist(f)) {
+                            dPrint("Read properties file "+f);
+                            cacheProps.load(ss.getFileInputStream(f));
+                        }
+                    }
+                }
+            }
+            final String factoryClassName = cacheProps.getProperty(factoryId);
+
+            if (factoryClassName != null) {
+                dPrint("found in $java.home/jaxp.properties, value=" + factoryClassName);
+                return newInstance(type, factoryClassName, null, true);
+            }
+        }
+        catch (Exception ex) {
+            if (debug) ex.printStackTrace();
+        }
+
+        // Try Jar Service Provider Mechanism
+        T provider = findServiceProvider(type);
+        if (provider != null) {
+            return provider;
+        }
+        if (fallbackClassName == null) {
+            throw new FactoryConfigurationError(
+                "Provider for " + factoryId + " cannot be found");
+        }
+
+        dPrint("loaded from fallback value: " + fallbackClassName);
+        return newInstance(type, fallbackClassName, null, true);
+    }
+
+    /*
+     * Try to find provider using the ServiceLoader API
+     *
+     * @param type Base class / Service interface  of the factory to find.
+     *
+     * @return instance of provider class if found or null
+     */
+    @SuppressWarnings("nullness")    // null return is acceptable as return value. 
```

Comment:
I don't understand this comment.  What warning is being suppressed, and how do we know that cannot possibly cause a problem elsewhere?

Reply:
Without this, while running checker the following mentioned above error comes, which doesn't allow null to be returned in this case. But according to specification null is allowed, so I think this suppression would not be causing any problem elsewhere. 
The comment is about acceptability of null value as return which is producing error. 

---
But I think considering the discussion from pr #1849 will be much better approach here and are valid annotations in this case.